ts/Series/FlowMap/FlowMapSeries.ts

```diff
@@ -0,0 +1,834 @@
+             */
+            width: '40%'
+        },
+
+        /**
+         * Maximum flow map links weight. It will automatically
+         * size between the minWeight and maxWeight.
+         *
+         * @since   next
+         */
+        maxWeight: 25,
+
+        /**
+         * Minimum flow map links weight. It will automatically
+         * size between the minWeight and maxWeight.
+         *
+         * @declare Highcharts.SeriesFlowMapSeriesOptionsObject
+         * @since   next
+         */
+        minWeight: 5,
+
+        tooltip: {
+            /**
+             * The HTML of the tooltip header line. Variables are enclosed by
+             * curly brackets. Available variables are point.key, series.name,
+             * series.color and other members from the point and series
+             * objects. The point.key variable contains the category name, x
+             * value or datetime string depending on the type of axis. For
+             * datetime axes, the point.key date format can be set using
+             * tooltip.xDateFormat.
+             *
+             * TODO: check why it's not generated in API
+             */
+            headerFormat: '<span style="font-size: 10px">{series.name}</span><br/>',
+            pointFormat: '{point.options.from} \u2192 {point.options.to}: <b>{point.weight}</b><br/>'
+        },
+
+        /**
+         * The weight of the link determines how thick it will be compared to
+         * other weights.
+         *
+         * @type    {number}
+         * @since   next
+         */
+        weight: void 0
+
+    } as FlowMapSeriesOptions);
+
+    /* *
+     *
+     *  Static Function
+     *
+     * */
+
+    /**
+     * Get vector length.
+     * @private
+     */
+    public static getLength(x: number, y: number): number {
+        return Math.sqrt(x * x + y * y);
+    }
+
+    /**
+     * Return a normalized vector.
+     * @private
+     */
+    public static normalize(x: number, y: number): Array<number> {
+        const length = FlowMapSeries.getLength(x, y);
+        return [x / length, y / length];
+    }
+
+    /**
+     * Return an SVGPath for markerEnd.
+     * @private
+     */
+    public static markerEndPath(
+        lCorner: [number, number],
+        rCorner: [number, number],
+        topCorner: [number, number],
+        options: MarkerEndOptions
+    ): SVGPath {
+        let width = options.width || 0;
+        const type = options.markerType || 'arrow',
+            [edgeX, edgeY] = this.normalize(
+                rCorner[0] - lCorner[0],
+                rCorner[1] - lCorner[1]
+            );
+
+        if (typeof width === 'string') {
+            const percentage = parseInt(width, 10);
+            width = this.getLength(
+                rCorner[0] - lCorner[0],
+                rCorner[1] - lCorner[1]) * percentage / 100;
+        }
+
+        let path: SVGPath = [];
+
+        // For arrow head calculation
+        if (type === 'arrow') {
+
```

Comment:
By looking at the other `if` statements in this file, I think this line break is unnecessary here.